nav2_costmap_2d/include/nav2_costmap_2d/costmap_filters/speed_filter.hpp

```diff
@@ -0,0 +1,117 @@
+/*********************************************************************
+ *
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020 Samsung Research Russia
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the <ORGANIZATION> nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Alexey Merzlyakov
+ *********************************************************************/
+
+#ifndef NAV2_COSTMAP_2D__SPEED_FILTER_HPP_
+#define NAV2_COSTMAP_2D__SPEED_FILTER_HPP_
+
+#include "nav2_costmap_2d/costmap_filters/costmap_filter.hpp"
+
+#include "nav_msgs/msg/occupancy_grid.hpp"
+#include "nav2_msgs/msg/costmap_filter_info.hpp"
+#include "nav2_msgs/msg/speed_limit.hpp"
+
+#include <memory>
+
+namespace nav2_costmap_2d
+{
+
+class SpeedFilter : public CostmapFilter
+{
+public:
+  SpeedFilter();
+
+  void initializeFilter(
+    const std::string & filter_info_topic);
+
+  void process(
+    nav2_costmap_2d::Costmap2D & master_grid,
+    int min_i, int min_j, int max_i, int max_j,
+    const geometry_msgs::msg::Pose2D & pose);
+
+  void resetFilter();
+
+  bool isActive();
+
+protected:
+  /**
+   * @brief: Name of speed limiting topic
+   */
+  std::string speed_limit_topic_;
+
+private:
```

Comment:
Why not make them all protected? It would be helpful later if we write tests deriving from this class if we can access these methods.

Reply:
Moved `transformPose()`, `worldToMask()` and `getMaskData()` methods to protected in https://github.com/ros-planning/navigation2/pull/2074/commits/3465c90415afd11418f98fc01bbc9386afa853ea#diff-db4cc04c6df9717814a17beefe3d9013c83753db07ef4a6bfc47e4ce97240d31